noir-projects/noir-contracts/contracts/token_with_refunds_contract/src/main.nr

```diff
@@ -469,35 +469,41 @@ contract TokenWithRefunds {
         // to the user in the `complete_refund(...)` function.
         storage.balances.sub(user, U128::from_integer(funded_amount)).emit(encode_and_encrypt_note_with_keys(&mut context, user_ovpk, user_ivpk, user));
 
-        // 4. We create the partial notes for the fee payer and the user.
+        // 4. Now we "manually" compute the slots (by setting the slots we insert the notes to the balances map under
```

Comment:
I think computing the slot here manually is pretty ugly. If we decide to make partial notes a first class citizen. I think it would make sense to modify the state vars such that they return the note hiding point instead of immediately emitting the note hashes. Or maybe just having separate for 2 flows on the state vars (one returning note hiding point, another directly emitting). 

What does the reviewer think?